### ISSUE TITLE
把vit-b-16模型转成ONNX格式，报：Exporting the operator 'aten::unflatten' to ONNX opset version 13 is not supported.

### ISSUE THREAD

**Issue by JustinZou1** [OFA-Sys/Chinese-CLIP#112]
1.执行下面的脚本把clip_cn_vit-b-16.pt模型转成ONNX格式：
<img width="780" alt="1684237292490" src="https://github.com/OFA-Sys/Chinese-CLIP/assets/128281676/2f9ebdf8-6326-42ac-8242-b26d1beb1e98">

2.报如下错误：
<img width="922" alt="1684237544309" src="https://github.com/OFA-Sys/Chinese-CLIP/assets/128281676/6a3d8ce2-941b-4485-87be-3801eb649b57">

3.文本模型是可以正常转换成onnx。img模型是失败了。
<img width="518" alt="1684237578070" src="https://github.com/OFA-Sys/Chinese-CLIP/assets/128281676/aeff91f8-63fe-484d-bd5b-8039448e98f0">

4.对应cn_clip/deploy/pytorch_to_onnx.py的代码是如下导出命令
<img width="787" alt="1684237658775" src="https://github.com/OFA-Sys/Chinese-CLIP/assets/128281676/310db01a-ad56-4336-8ac1-5a446c966a9d">

**Comment by yangapku** [comment]
您好，我们初步判断，这种情况一般是onnx version的版本低了，安装版本的onnx还不支持这个算子，可以试着把optset-version调高一点试试

**Comment by JustinZou1** [comment]
@yangapku ,谢谢您的回复。
您好，这个是我安装的库和版本，具体是需要把哪个组件升级到哪个版本？谢谢。
(chineseclip) ubuntu@justingpu1:~$ pip list
Package                  Version
------------------------ ------------
certifi                  2022.12.7
charset-normalizer       2.1.1
cmake                    3.25.0
cn-clip                  1.5.1
coloredlogs              15.0.1
filelock                 3.9.0
flatbuffers              23.5.9
fsspec                   2023.5.0
huggingface-hub          0.14.1
humanfriendly            10.0
idna                     3.4
Jinja2                   3.1.2
joblib                   1.2.0
lit                      15.0.7
lmdb                     1.3.0
MarkupSafe               2.1.2
mpmath                   1.2.1
networkx                 3.0
numpy                    1.24.1
nvidia-cublas-cu12       12.1.3.1
nvidia-cuda-runtime-cu12 12.1.105
nvidia-cudnn-cu12        8.9.1.23
onnx                     1.14.0
onnxconverter-common     1.13.0
onnxmltools              1.11.2
onnxruntime-gpu          1.14.1
packaging                23.1
Pillow                   9.3.0
pip                      23.0.1
protobuf                 4.23.0
PyYAML                   6.0
requests                 2.28.1
safetensors              0.3.1
scikit-learn             1.2.2
scipy                    1.10.1
setuptools               66.0.0
six                      1.16.0
skl2onnx                 1.14.1
sympy                    1.11.1
tensorrt                 8.6.1
tensorrt-bindings        8.6.1
tensorrt-libs            8.6.1
threadpoolctl            3.1.0
timm                     0.9.2
torch                    2.0.1+cu118
torchaudio               2.0.2+cu118
torchvision              0.15.2+cu118
tqdm                     4.65.0
triton                   2.0.0
typing_extensions        4.4.0
urllib3                  1.26.13
wheel                    0.38.4
(chineseclip) ubuntu@justingpu1:~$

**Comment by JustinZou1** [comment]
也试了把 opset_version设置为14了，还是报一样的错误：
<img width="910" alt="1684478717233" src="https://github.com/OFA-Sys/Chinese-CLIP/assets/128281676/987bf522-5daa-4321-916c-eb5ff86cb16b">

**Comment by yangapku** [comment]
您好，看到您的pytorch版本较高（2.0.1），我们确实还没有验证过在此版本下deploy功能的可用性。最稳妥的建议，是建议您对齐我们deployment.md中的各项环境版本，再实验一下，应该是能够跑通的。

**Comment by yangapku** [comment]
或者您可以再试试调高opset-version到更高（应该是大于16的一个值），或者就需要考虑用手写替代这个算子了。

**Comment by JustinZou1** [comment]
试了，15,16,17,18都不行，报一样的错。在试19时，报如下错误，错误有点不一样。
<img width="916" alt="1684479686049" src="https://github.com/OFA-Sys/Chinese-CLIP/assets/128281676/87a221dc-2a41-4bd7-a6de-fc326fc73f66">

**Comment by JustinZou1** [comment]
您好，也重新执行pip install -r requirements.txt，创建了一个新的conda env,还是报一样的错。重新创建的pytorch环境还是2.0.1+cu118，自动会安装最新的torch。是需要手工指定安装哪个版本的pytorch环境？

**Comment by yangapku** [comment]
您好，ONNX部署需要考虑的环境有CUDA（nvidia-smi显示的机器cuda库版本，不是conda cudatoolkit库版本）、CUDNN、ONNX和Pytorch的版本，您可以整体对齐到我们[deployment.md](https://github.com/OFA-Sys/Chinese-CLIP/blob/master/deployment.md)的版本上，需要重新安装对应版本的上述库，这样应该是我们充分验证过能跑通的。或者，如果您想基于自己的CUDA库版本配置环境，就需要参考deployment.md给出的说明，自己查找下对应应该安装的CUDNN、ONNX和Pytorch的版本哈。

**Comment by JustinZou1** [comment]
@yangapku 谢谢您的帮忙，我严格按照文档中的版本，重新部署了下，是可以正常转。
